system-design/app-interoperability/workload-orchestration-agent.md

```diff
@@ -83,9 +83,214 @@ The following shall be configurable to ensure compliance with local operations.
+| properties       | map[string][interface{}] | Y              | A dictionary element specifying the component packages's deployment details. See the [Component Properties](#component-properties) section below.|
+
+
+**Component Properties**
+
+The expected properties for the suppported deployment types are indicated below.
+
+- Properties for `helm.v3` components
+
+    | Attribute        | Type            | Required?       | Description     |
+    |------------------|-----------------|-----------------|-----------------|
+    | repository       | string          | Y               | The URL indicating the helm chart's location.|
+    | revision         | string          | Y               | The helm chart's full version.|
+    | wait             | bool            | N               | If `True`, indicates the device MUST wait until the helm chart has finished installing before installing the next helm chart. The default is `True`. The Workload Orchestration Agent MUST support `True` and MAY support `False`. Only applies if multiple `helm.v3` components are provided.|
+    | timeout        | string          | N    | Describes the timeout period to wait before moving to th enext component in the deployment specification.|
+
+- Properties for `docker-compose` components
+
+    > **Investigation Needed**: We need to have more discussion about how docker-compose should be handled and what is required here.
+    > We also need to determine if there is a version of docker-compose that needs to be specified. The docker compose schema [version has been
+    > deprecated](https://github.com/compose-spec/compose-spec/blob/master/spec.md#version-and-name-top-level-elements) so it's not clear what we would even use for this if we wanted to.
+
+    | Attribute        | Type            | Required?       | Description     |
+    |------------------|-----------------|-----------------|-----------------|
+    | packageLocation  | string          | Y               | The URL indicating the Docker Compose package's location. |
+    | keyLocation      | string          | N               | The public key used to validated the digitally signed package. It is highly recommend to digitally sign the package. When signing the package PGP MUST be used.|
+    | wait             | bool            | N               | If `True`, indicates the device MUST wait until the Docker Compose file has finished starting up before starting the next Docker Compose file. The default is `True`. The Workload Orchestration Agent MUST support `True` and MAY support `False`. Only applies if multiple `docker-compose` components are provided.|
+    | timeout        | string          | N    | Describes the timeout period to wait before moving to th enext component in the deployment specification.|
+
+**Parameter Attributes**
```

Comment:
We should probably make sure our descriptions are aligned here or figure out how we want to handle cross references. The descriptions for the parameter elements in [my branch](https://github.com/margo/specification/blob/pdp/app-description-parameters/system-design/app-interoperability/application-package-definition.md) are different than what you have.

Reply:
Agreed, not sure this will be the first time this occurs. A good strategy should be found for this instance and future.